docs/index.md

```diff
@@ -0,0 +1,187 @@
+# Regula
+
+## Overview
+
+The "Regula" project addresses the problem caused by the accumulation of simple but recurrently changing logic due to business requirements. This introduces strains on the development cycle by reducing the velocity. In concrete terms, developers can find themselves always chasing the business rather than focusing on core logic which defines the business.
+
+A solution to his kind of problems is a "Business Rules engine" or "Production Rule System". It solves it by providing a way to extract logic by creating "rules" that dictates a result under precise conditions. 
+
+Such rules are ordered and composed of a condition and an action - simplistically, it's a bunch of ordered if-then statements. [Learn more about Rule-Engines](https://martinfowler.com/bliki/RulesEngine.html).
+
+What really makes this approach shine is that stakeholders can change the rules themselves, without having to wait for developers to implement it. This enables much faster development cycles.
+
+Example: Let's say there's a ride-sharing company that needs to adjust the radius in which it can match drivers and passengers. This changes a lot, depending on various real life events:
+
+```
+Given I want to know "What radius should I match driver and passenger with" in my code 
+And I don't want to hardcode the logic
+When I query the ruleset named "/ride-marketplace/radius"
+And that I submit "product", "driver-status" and "city"
+Then I get a radius value based on rules that have been defined by product owners
+
+---
+
+radius = 
+  // Product people can update these rules on their own
+  // Codebase using Regula will receive those updates live
+  if city == "france-paris" AND driver-status == "gold" then 3.0
+  if city == "france-paris" then 2.0
+  if city == "italy-milan" then 1.5
+  // catch all clause, acts as a default
+  if true then 1.0
+```
+
+### How does it help being productive
+
+So once developers and stakeholders have agreed on what simple logic will be extracted, it is possible to split the ownership in two:
+
+- "when and how it's being used": developers own the context. Developers are responsible on how that logic is being used inside the codebase.
+- "how is it computed ": stakeholders own the behaviour. Stakeholders can update the logic itself in total autonomy. 
+
+
+The following diagrams highlights this split, with a **ruleset** that computes a radius. 
+
+![Before/After schema](./before_after.png)
+
+
+| Before                                                            | After |
+|-------------------------------------------------------------------|-------|
+| 1⃣ Developers and stakeholders agrees on a given solution   | 1⃣ Same  |
+| 2⃣ Stakeholders write specifications                        | 2⃣ Stakeholders and developers agrees on how what parameters the ruleset should take as well as its return type: **developers create the ruleset on Regula.**
```

Comment:
`Stakeholders and developers agrees on how what parameters the ruleset should take as well as its return type: **developers create the ruleset on Regula.**`
-> 
```
- Stakeholders and developers agree on a **signature**: what parameters the ruleset should take as well as its return type
- **Developers create the ruleset on Regula.**
```